### DISCUSSION TITLE
Open pdf from google drive

### DISCUSSION THREAD

**Discussion by milperks** [Pattern-Projector/pattern-projector#329] [Ideas]
Would you consider adding different options for where to open the pdf from? It would be particularly helpful if pdfs could be opened from google drive.

Thank you for this project and your commitment - it's amazing!

**Comment by courtneypattison** [comment]
Thank you for reaching out @milperks ! I've considered adding Google Drive support since a lot of people--myself included--use it to store patterns. However, I have the drive app installed and it's as easy as loading a file from my file system so I'm happy enough with that for the time being. I'd rather not have to deal with authentication issues, lol.

**Reply by milperks** [reply]
That's fair, I imagine authentication would add a lot of complexity! I'll look into the drive app.

**Comment by TLCary** [comment]
The HY320 (Magic Cube) Projector (very common) is Android based and has the Chrome web browser.  But, Open PDF goes to the Default upload folder - which is "Music" - and it doesn't let me navigate from there.  Unfortunately the Default download folder is "Downloads".   And it ties our hands when it comes to moving files around.   Being able to open from another web source would be ideal.   It's completely a HY320 limitation - but if i could get it to work with Pattern-Projector it would be ideal.

**Reply by courtneypattison** [reply]
Sorry @TLCary, I still don't plan on adding Drive support. I'd suggest hooking your Magcubic up to a computer/tablet/phone to access drive files. I did add "Open with" functionality on the [beta site](beta.patternprojector.com), so if you have a computer hooked up to your projector, you can use the drive app to open a file.
<img width="1091" alt="Screenshot 2025-02-26 at 12 07 12 PM" src="https://github.com/user-attachments/assets/ea871034-cbed-490f-91fb-8341e9825ce5" />

If you really want to use just the projector, maybe try using a file manager to cut and paste files into the directory you have access to with PP. Also, Chrome on Android has a pretty serious bug so make sure you use Firefox on Android!
